controller/rest/remote_repository.go

```diff
@@ -0,0 +1,186 @@
+package rest
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+	"github.com/neuvector/neuvector/controller/api"
+	"github.com/neuvector/neuvector/controller/common"
+	log "github.com/sirupsen/logrus"
+)
+
+func handlerRemoteRepositoryPost(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.WithFields(log.Fields{"URL": r.URL.String()}).Debug()
+	defer r.Body.Close()
+
+	acc, login := getAccessControl(w, r, "")
+	if acc == nil {
+		return
+	} else if !acc.Authorize(&api.RemoteRepository{}, nil) {
+		restRespAccessDenied(w, login)
+		return
+	}
+
+	body, _ := ioutil.ReadAll(r.Body)
+	var remoteRepository api.RemoteRepository
+	err := json.Unmarshal(body, &remoteRepository)
+	if err != nil {
+		msg := fmt.Sprintf("Could not unmarshal request body: %s", err.Error())
+		restRespErrorMessage(w, http.StatusBadRequest, api.RESTErrInvalidRequest, msg)
+		return
+	} else if !isObjectNameValid(remoteRepository.Nickname) {
```

Comment:
In 5.3 we only allow one entry with alias as "default".
Please check whether the Alias field is "default". Return error if it's not.

Reply:
I don't think this should be enforced by the back-end. As it currently stands, the API supports multiple destinations, and it has already been implemented that way.

If the web UI front-end will only support a single destination for now, it does not have to make the `nickname` field editable by the user, and only send requests using the `default` alias behind the scenes.